Reg 19

arianaolson419 opened the pull request:
https://olinjs.atlassian.net/browse/REG-19

arianaolson419 committed:
Create container for Student Home Page

In anticipation of connecting login functionality with app, it is necessary
to provide the home page with user-specific information. The idea is that
the state will update when a user logs in.

arianaolson419 committed:
Add user prop to component

Because there is now a comtainer for this component, it now needs to
be able to accept a user object a a property. Elements of that object
are passed to children as needed.

arianaolson419 committed:
Change name of prop to reflect user object key

arianaolson419 committed:
Merge branch 'master' of https://github.com/olin/register into REG-19

arianaolson419 committed:
Change action to match format of others

arianaolson419 committed:
Add actions for login form updates

These are taken from the redux tutorial code in class. Part of an effort
to reduxify the login page.

arianaolson419 committed:
Delete unused example file

arianaolson419 committed:
Convert component to presentational component

arianaolson419 committed:
Create container component connecting to LocalLogin presentational component

arianaolson419 committed:
Create reducer for local login

arianaolson419 committed:
Combine all separate reducers

arianaolson419 committed:
Add Register actions

arianaolson419 committed:
Change initial state to correct shape

arianaolson419 committed:
Create reducers for register component

arianaolson419 committed:
Fix typos

arianaolson419 committed:
Import correct actions

arianaolson419 committed:
Create Register contatiner component

arianaolson419 committed:
Change action to more descriptive name

arianaolson419 committed:
Add another dispatch

arianaolson419 committed:
Convert to redux

arianaolson419 committed:
Change name to match corresponding action

arianaolson419 committed:
Fix linter errors

arianaolson419 committed:
Add actions for login backend interaction

arianaolson419 committed:
Fix linting errors

arianaolson419 committed:
Install redux-thunk

arianaolson419 committed:
Use jquery

arianaolson419 committed:
Play with making asynch log in work

arianaolson419 committed:
add argument to specify passport fields

arianaolson419 committed:
Remove jquery script

arianaolson419 committed:
Change login after register code

The passport.authenticate function did not work the way it was being
implemented in the /register route. By using req.login instead,
the user is successfully logged in after registering.

arianaolson419 committed:
Add register action and fix login action bugs

arianaolson419 committed:
Incorporate onRegister and fix bugs in updating state

arianaolson419 committed:
Incorporate register action and switch to mergeProps

In order to access the current state for registration purposes, it was
necessary to use mergeProps as an argument to the connect
fuction.

arianaolson419 committed:
Incorporate login function and use mergeProps

Merge props was need to access elements of the state that are passed to the
login fuction as arguments.

arianaolson419 committed:
Pull username from state subobject

arianaolson419 committed:
Apply middleware for thunks

arianaolson419 committed:
Change shape of state and add receive user reducer

arianaolson419 committed:
Delete blank line

arianaolson419 committed:
Remove undefined prop from propTypes

arianaolson419 committed:
Remove unused function

arianaolson419 committed:
Merge branch 'REG-19' of https://github.com/olin/register into REG-19

arianaolson419 committed:
Fix linter errors

arianaolson419 committed:
Transfer router fixes from REG-20

arianaolson419 committed:
Change name for clarity

arianaolson419 committed:
Fix linter errors

arianaolson419 committed:
Fix router errors, still cannot redirect to /login

arianaolson419 committed:
Remove server-side redirect logic

Because BrowserRouter redirects on the client side, server redirects
do not render a new page. All redirects currently happen on the client side
using React Router.

arianaolson419 committed:
Add redirect logic

All redirecting happens on the client side now. When a user is not
logged in (determined by loggedIn flag in the state), the page redirects
to the /login route. If a user is logged in, the login page redirects
to the home page.

arianaolson419 committed:
Delete print statement and add break before button

arianaolson419 committed:
Delete print statement

arianaolson419 committed:
Add loggedIn flag for redirect logic

arianaolson419 committed:
Render AppContainer

arianaolson419 committed:
Create container to pass loggedIn flag to props for redirect logic

logandavis commented:
I don't think I'm fully on board with how you've split up your actions, props, and state for logging in vs. registering. I'd prefer more descriptive names than "USERNAME" vs. "NEW_USERNAME", and I'd prefer that the props and state refer to those attributes in the same way that the actions do. I think it might be good to spend some time thinking about state shape. Do the Login and Register usernames that you keep in state need to be different? Do they need different actions? If so, can you name those state fields descriptively?

logandavis reviewed (changes requested):
I'd like more consistent names / organization for the state fields and the actions associated with the Register and Login components.

logandavis commented:
Your commit messages are looking good! I notice that you have some commits with names like "Fix linter errors". Before you merge, I bet it'd be a good learning experience to use `git rebase -i` to clean up the commit history a little. That'd also help you fix the merge conflicts with master that you currently have.

yunhsincynthiachen commented on src/components/RegisterNewUser.jsx:
```diff
@@ -1,108 +1,55 @@
+      />
+      <br />
+      <input
+        placeholder="Confirm Password"
```
It doesn't really seem like we're doing anything really with "confirm password" field (ideally, it would do a check between this field and the new password field, and alert if not the same). Let's remove this for now, and tag this as future work, since it's not super crucial to implement for now.

yunhsincynthiachen commented on src/components/LoginLocal.jsx:
```diff
@@ -1,65 +1,39 @@
-import React from 'react';
-import $ from 'jquery';
+import React, { PropTypes } from 'react';
```
merge with master to get the style stuff in place

yunhsincynthiachen commented on src/components/RegisterNewUser.jsx:
```diff
@@ -1,108 +1,55 @@
-*/
-import React from 'react';
-import $ from 'jquery';
+import React, { PropTypes } from 'react';
```
merge with master to get the style stuff in place

yunhsincynthiachen commented on src/actions/actions.jsx:
```diff
@@ -1,5 +1,65 @@
+    };
+    $.post('/login', data)
+      .done(response => dispatch(receiveUser(response)))
+      .fail((err, status) => console.log(err, status));
```
Is there any way we can have an alert error pop up when the user doesn't have an account/password is incorrect?

yunhsincynthiachen commented on src/actions/actions.jsx:
```diff
@@ -1,5 +1,65 @@
+
+// Register Component
+export const updateNewUsername = username => ({
+  type: 'UPDATE_NEW_USERNAME',
```
maybe have "REGISTER", instead of "NEW", more specific to what your reducer is?

yunhsincynthiachen commented on src/actions/actions.jsx:
```diff
@@ -1,5 +1,65 @@
+// Register Component
+export const updateNewUsername = username => ({
+  type: 'UPDATE_NEW_USERNAME',
+  username,
```
I do agree with Logan, in that it would be a good idea to have keys that are more specific to what the component is.

yunhsincynthiachen reviewed (changes requested):
Everything looks awesome, and really detailed. I have a couple changes, addressing some stuff Logan talked about, as well as removing unnecessary components. Also, master would be good to merge now.

arianaolson419 committed:
Remove all confirm password related code

arianaolson419 committed:
Delete duplicate

arianaolson419 committed:
Change all 'updateNew's to 'updateRegister's

yunhsincynthiachen commented:
Good changes! Still  need to merge with master to make sure nothing will break, especially because Taylor's stuff is merged.

arianaolson419 committed:
Merge branch 'master' of https://github.com/olin/register into REG-19

Conflicts:
	routes/index.js
	src/components/App.jsx
	src/components/LoginLocal.jsx
	src/components/LoginPage.jsx
	src/components/RegisterNewUser.jsx
	src/components/StudentHome.jsx
	src/index.jsx

Halliax commented:
This branch is no longer suffering from merge conflicts, but unfortunately login state doesn't persist. We've been sort of blocked on that problem for this afternoon

billmwong commented:
We're putting the persisting login state bug in another ticket (REG-60), and going forward with this PR as-is.

- [x] Delete code that was used for testing so that this is production/master ready
- [x] Resolve conflicts with master again

arianaolson419 committed:
Remove login redirect logic

Until we can make the state persist, having everything redirecting to the
login page is a huge pain

arianaolson419 committed:
Change settings button link back to settings route

billmwong commented:
Resolve conflicts with master again

arianaolson419 committed:
Merge branch 'master' of https://github.com/olin/register into REG-19

Conflicts:
	src/components/NavPanel.jsx

arianaolson419 committed:
Add redirect logic for login page

arianaolson419 committed:
Delete persist code

arianaolson419 committed:
Make register names match in reducer and actions

yunhsincynthiachen commented:
Looks great! I tested the register and login functionalities, which work. Persist after redirect doesn't work, but @arianaolson419 is working on this in a different branch!
